utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -491,31 +391,122 @@ fn extract_code(spec: Box<dyn ChainSpec>) -> sc_cli::Result<(StorageKey, Storage
 	Ok((code_key, code))
 }
 
+/// Get the hash type of the generic `Block` from a `hash_str`.
+pub(crate) fn hash_of<Block: BlockT>(hash_str: &str) -> sc_cli::Result<Block::Hash>
+where
+	Block::Hash: FromStr,
+	<Block::Hash as FromStr>::Err: Debug,
+{
+	hash_str
+		.parse::<<Block as BlockT>::Hash>()
+		.map_err(|e| format!("Could not parse block hash: {:?}", e).into())
+}
+
 /// Check the spec_name of an `ext`
 ///
 /// If the version does not exist, or if it does not match with the given, it emits a warning.
-async fn check_spec_name<Block: BlockT + serde::de::DeserializeOwned>(
+pub(crate) async fn ensure_matching_spec_name<Block: BlockT + serde::de::DeserializeOwned>(
 	uri: String,
 	expected_spec_name: String,
 ) {
-	let expected_spec_name = expected_spec_name.to_lowercase();
 	match remote_externalities::rpc_api::get_runtime_version::<Block, _>(uri.clone(), None)
 		.await
 		.map(|version| String::from(version.spec_name.clone()))
 		.map(|spec_name| spec_name.to_lowercase())
 	{
 		Ok(spec) if spec == expected_spec_name => {
-			log::debug!("found matching spec name: {:?}", spec);
+			log::debug!(target: LOG_TARGET, "found matching spec name: {:?}", spec);
 		},
 		Ok(spec) => {
-			log::warn!(
+			panic!(
```

Comment:
Is there normally a spec name mismatch because we use `Development locally?

Reply:
not anymore: if you see the code, now I get the local spec version from the runtime. In the past, `--chain polkadot-dev` was `development`, but now it should be `polkadot`. 

Nonetheless, it might be useful for example for the bot to be able to run this in substrate against polkadot state. So I will add a flag to disable this check.